### PR TITLE
Avoid fetching external DTD/ schema

### PR DESCRIPTION
**Pull request by mskdeepak-oracle**

### CODE REVIEW

**Comment by mskdeepak-oracle** [comment]
@glassfishrobot Run CI tests please

**Comment by mskdeepak-oracle** [comment]
@glassfishrobot Run CI tests please

**Review by honghzzhang — DISMISSED** [review]
Changes look ok to me.

File: appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/AppClientContainerBuilder.java [hunk 1]
```diff
@@ -63,6 +63,7 @@
 import org.glassfish.appclient.client.acc.config.TargetServer;
 import org.glassfish.appclient.client.acc.config.util.XML;
 import org.glassfish.enterprise.iiop.api.GlassFishORBHelper;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/AppClientContainerBuilder.java]: This import is unused, can be removed

File: appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/Launchable.java [hunk 2]
```diff
@@ -56,6 +56,7 @@
 import javax.xml.stream.XMLStreamException;
 import org.glassfish.api.deployment.archive.ReadableArchive;
 import org.glassfish.hk2.api.ServiceLocator;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/Launchable.java]: Don't you need to replace the exception at line #189?

File: appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/MainClassLaunchable.java [hunk 3]
```diff
@@ -59,6 +59,7 @@
 import java.util.zip.ZipEntry;
 import org.glassfish.api.deployment.archive.ReadableArchive;
 import org.glassfish.hk2.api.ServiceLocator;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/MainClassLaunchable.java]: Unused import can be removed.

File: appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/UndeployedLaunchable.java [hunk 4]
```diff
@@ -62,6 +62,7 @@
 import org.glassfish.api.deployment.archive.ReadableArchive;
 import org.glassfish.deployment.common.RootDeploymentDescriptor;
 import org.glassfish.hk2.api.ServiceLocator;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/UndeployedLaunchable.java]: There are still some methods throwing the wrapped exception. Do we need it?

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/AppClientArchivist.java [hunk 5]
```diff
@@ -56,6 +56,7 @@
 
 import org.jvnet.hk2.annotations.Service;
 import org.glassfish.hk2.api.PerLookup;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/AppClientArchivist.java]: Unused import

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/ApplicationArchivist.java [hunk 6]
```diff
@@ -67,6 +67,7 @@
 
 import org.jvnet.hk2.annotations.Service;
 import org.glassfish.hk2.api.PerLookup;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/ApplicationArchivist.java]: Unused import

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/ApplicationFactory.java [hunk 7]
```diff
@@ -55,6 +55,7 @@
 import javax.inject.Inject;
 
 import org.jvnet.hk2.annotations.Service;
+import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
```
Comment (yaminikb) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/archivist/ApplicationFactory.java]: Please remove the unused import in all files.

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/LocalStrings.properties [hunk 8]
```diff
@@ -42,5 +42,6 @@ enterprise.deployment.node.invaliddeploymentdescriptors=Invalid Deployment Descr
 
 enterprise.deployment.node.invalidvalue=Invalid value for a tag under {0} : {1}
 invalid.root.element={0} Element [{1}] is not a valid root element
+invalid.schema=Requested schema not in local repository, check for typos in the descriptor file
```
Comment (yaminikb) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/LocalStrings.properties]: Please rephrase the above message as

"Requested schema is not found in local repository, please ensure that there are no typos in the XML namespace declaration."

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/SaxParserHandler.java [hunk 9]
```diff
@@ -242,25 +248,22 @@ public InputSource resolveEntity(String publicID, String systemID) throws SAXExc
                     } else {
                       fileName = getSchemaURLFor(systemID.substring(systemID.lastIndexOf('/')+1));                    
                     }
-                    // if this is not a request for a schema located in our repository, 
-                    // let's hope that the hint provided by schemaLocation is correct
+                    // if this is not a request for a schema located in our repository, we fail the deployment
                     if (fileName==null) {
-                        fileName = systemID;
+                        throw new SAXException(localStrings.getLocalString("invalid.schema", "Requested schema not in local repository, check for typos in the descriptor file"));
```
Comment (yaminikb) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/SaxParserHandler.java]: Fix the message as mentioned in previous comment.

**Review by yaminikb — CHANGES_REQUESTED** [review]

**Comment by mskdeepak-oracle** [comment]
@glassfishrobot Run CI tests please

**Comment by mskdeepak-oracle** [comment]
@glassfishrobot Run CI tests please

**Comment by mskdeepak-oracle** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]

**Pull request merged**